src/unit.rs

```diff
@@ -281,9 +282,12 @@ fn match_cookies(jar: &CookieJar, domain: &str, path: &str, is_secure: bool) ->
             let name = c.name().to_string();
             let value = c.value().to_string();
             let nameval = Cookie::new(name, value).encoded().to_string();
-            Header::new("Cookie", &nameval)
+            nameval
         })
-        .collect()
+        .collect::<Vec<_>>()
+        .join(";");
+
+    Header::new("Cookie", &cookie)
```

Comment:
I don't *think* that setting an empty cookie header is a bad thing but it may also not be desired?

Reply:
I agree, we should avoid sending the cookie header if there are no cookies. How about making this function return an `Option` too?

Reply:
Done :)